### PR TITLE
Improved support for x-go-type on compound objects

### LINKED ISSUE #1719: External x-go-type imports not resolved for arrays and references
# Problem statement
Ok, after few days of debugging I give up. Problem with imports using `x-go-type` still exists after my changes, is seems that it fixed only imports when flat object definition is used.

File: **get_user_obj_ref_responses.go** - FAIL
```go
import (
	"net/http"

	"github.com/go-openapi/runtime"

        // myalias3 "github.com/user/externalrepo/pkg/models" <--- MISSING!
)
```
***With swagger 0.16.0 same result.***

File: **get_user_obj_flat_responses.go** - CORRECT
```go
import (
	"net/http"

	"github.com/go-openapi/runtime"

	myalias4 "github.com/user/externalrepo/pkg/models"
)
```
***With swagger 0.16.0 import MISSING***

File: **get_users_arr_ref_responses.go** - FAIL
```go
import (
	"net/http"

	"github.com/go-openapi/runtime"

	// myalias1 "github.com/user/externalrepo/pkg/models" <-- MISSING!
)
```
***With swagger 0.16.0 same result.***

File: **get_users_arr_flat_responses.go** - FAIL
```go
import (
	"net/http"

	"github.com/go-openapi/runtime"
        // myalias2 "github.com/user/externalrepo/pkg/models" <-- MISSING!
)
```
***With swagger 0.16.0 same result.***

## Swagger specification
```yml
swagger: '2.0'
info:
  title: 'x-go-type issues'
  version: '1.0'
paths:
  /users_arr_ref:
    get:
      responses:
        200:
          description: OK
          schema:
            $ref: '#/definitions/UsersArrayRef'
  /users_arr_flat:
    get:
      responses:
        200:
          description: OK
          schema:
            $ref: '#/definitions/UsersArrayFlat'
  /user_obj_ref:
    get:
      responses:
        200:
          description: OK
          schema:
            $ref: '#/definitions/UserObjectRef'
  /user_obj_flat:
    get:
      responses:
        200:
          description: OK
          schema:
            $ref: '#/definitions/UserObjectFlat'
definitions:
  User:
    type: object
    properties:
      name:
        type: string
      age:
        type: integer
  UsersArrayRef:
    type: array
    items:
       $ref: '#/definitions/User'
    x-go-type:
      import:
        alias: myalias1
        package: 'github.com/user/externalrepo/pkg/models'
      type: ExtUsers
  UsersArrayFlat:
    type: array
    items:
      type: object
      properties:
        name:
          type: string
        age:
          type: integer
    x-go-type:
      import:
        alias: myalias2
        package: 'github.com/user/externalrepo/pkg/models'
      type: ExtUsersFlat
  UserObjectRef:
    $ref: '#/definitions/User'
    x-go-type:
      import:
        alias: myalias3
        package: 'github.com/user/externalrepo/pkg/models'
      type: ExtUser
  UserObjectFlat:
    type: object
    properties:
      name:
        type: string
      age:
        type: integer
    x-go-type:
      import:
        alias: myalias4
        package: 'github.com/user/externalrepo/pkg/models'
      type: ExtUserFlat
```

## Steps to reproduce
`swagger generate server -f swagger.yml`

## Environment
swagger version: master  
go version: 1.11
OS: Ubuntu 18.04

### PR DESCRIPTION
**Pull request by fredbi**
* fixes #1719
* regenerated bindata with github.com/kevinburke/go-bindata v3.13.0

Signed-off-by: Frederic BIDON <fredbi@yahoo.com>

### CODE REVIEW

File: fixtures/bugs/1719/pkg/models/user.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Code generated by go-swagger; DO NOT EDIT.
```
Comment (casualjim) [inline fixtures/bugs/1719/pkg/models/user.go]: how come this file is included?
wouldn't it be overwritten every test

Comment (fredbi) [reply fixtures/bugs/1719/pkg/models/user.go]: I  manually generated some sample models to use as "external packages" from the main fixture spec.
I may remove the "DO NOT EDIT" line if you found this misleading.

Comment (casualjim) [reply fixtures/bugs/1719/pkg/models/user.go]: I was just curious

Comment (fredbi) [reply fixtures/bugs/1719/pkg/models/user.go]: ok good

**Pull request merged**